### ISSUE TITLE
Feature Request: Add geom_raster in gheatmap()

### ISSUE THREAD

**Issue by paprikachan** [YuLab-SMU/ggtree#450]
Hi, 

I am trying to use ggtree and gheatmap to visualize the marker genes from different single cells, and I got the left plot. Obviously, the marker pattern between different clusters is not clear. Then, I hardcoded the gheatmap function by changing "geom_tile" to "geom_raster", I got the right plot. Cleary, the raster heatmap shows a more distinct pattern when the plot size is small. Currently, I override the gheatmap function to achieve this raster effect, it would be more convenient if gheatmap can choose to plot from geom_tile or geom raster. Thanks a lot.

![1636119885(1)](https://user-images.githubusercontent.com/5020011/140519987-de23d5c4-7eab-4c41-bdf9-177c28bf9436.png)

**Comment by xiangpin** [comment]
I don't think it is a good idea. because `geom_raster` is a special case, please refer to this [issue](https://github.com/tidyverse/ggplot2/issues/3025). And it only support `coord_cartesian`.
